go/database/mpt/tool/export_test.go

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"io"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/Fantom-foundation/Carmen/go/common"
+	"github.com/Fantom-foundation/Carmen/go/database/mpt"
+	cio "github.com/Fantom-foundation/Carmen/go/database/mpt/io"
+)
+
+func TestExport_CanBeInterrupted(t *testing.T) {
+	type testFuncs struct {
+		export   func(context.Context, string, io.Writer) error
+		createDB func(t *testing.T, dir string)
+		check    func(t *testing.T, dir string) error
+	}
+	tests := map[string]testFuncs{
+		"archive": {
+			export:   cio.ExportArchive,
+			createDB: createArchiveTrie,
+			check:    checkExportedArchive,
+		},
+		"live": {
+			export:   cio.Export,
+			createDB: createLiveTrie,
+			check:    checkExportedLiveDB,
+		},
+	}
+
+	for name, tf := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			dir := filepath.Join(t.TempDir(), name)
+			tf.createDB(t, dir)
+			ctx, cancel := context.WithCancel(context.Background())
+			catchInterrupt(ctx, cancel, time.Now())
+
+			success := make(chan any, 1)
+			writer := newMockWriter(t)
+
+			// first find number of writes
+			if err := tf.export(ctx, dir, writer); err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
+			}
+
+			// save max count and reset number of writes
+			maxCount := writer.numOfWrites
+			writer.numOfWrites = 0
+
+			go func() {
+				if err := tf.export(ctx, dir, writer); err != nil {
+					t.Errorf("unexpected error: %v", err)
+					return
+				}
+				close(success)
+			}()
+
+			time.Sleep(200 * time.Millisecond)
+			err := syscall.Kill(syscall.Getpid(), syscall.SIGINT)
+			if err != nil {
+				t.Error("failed to create a SIGINT signal")
+				return
+			}
+			select {
+			case <-success:
```

Comment:
I want to make this test deadlock-safe, thats why I introduced a timeout

Reply:
how can this deadlock? If there is a bug in sense the kill signal is not correctly propagated, the worse case is that the export will complete, but it will not deadlock.

Reply:
You are right. Thanks!